python/ray/data/datasource/image_datasource.py

```diff
@@ -148,11 +151,20 @@ def __init__(
             partitioning=partitioning,
             **reader_args,
         )
-        self._encoding_ratio = self._estimate_files_encoding_ratio()
-        meta_provider._set_encoding_ratio(self._encoding_ratio)
+        if isinstance(meta_provider, _ImageFileMetadataProvider):
+            self._encoding_ratio = self._estimate_files_encoding_ratio()
+            meta_provider._set_encoding_ratio(self._encoding_ratio)
+        else:
+            self._encoding_ratio = IMAGE_ENCODING_RATIO_ESTIMATE_DEFAULT
 
     def estimate_inmemory_data_size(self) -> Optional[int]:
-        return sum(self._file_sizes) * self._encoding_ratio
+        total_size = 0
+        for file_size in self._file_sizes:
+            # NOTE: check if file size is not None, because some metadata provider
+            # such as FastFileMetadataProvider does not provide file size information.
+            if file_size is not None:
+                total_size += file_size
+        return total_size * self._encoding_ratio
```

Comment:
total_size still may be 0 though? is that ok?

Reply:
Yes it's ok - this is default behavior for file based data sources - https://github.com/ray-project/ray/blob/master/python/ray/data/datasource/file_based_datasource.py#L418-L423 .